src/Catalyst.Core.Modules.Consensus/Deltas/Building/TransactionRetrieverStep.cs

```diff
@@ -0,0 +1,131 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using Catalyst.Abstractions.Consensus.Deltas;
+using Catalyst.Protocol.Transaction;
+using Dawn;
+using Nethermind.Core;
+using Serilog;
+
+namespace Catalyst.Core.Modules.Consensus.Deltas.Building
+{
+    internal sealed class TransactionRetrieverStep : IDeltaBuilderStep
+    {
+        [Todo(Improve.Refactor, "Introduce configuration for delta / voting for size")]
+        private const ulong DeltaGasLimit = 8_000_000;
+
+        [Todo(Improve.Refactor, "Introduce configuration for tx cost")]
+        private const ulong MinTransactionEntryGasLimit = 21_000;
+
+        private readonly ILogger _logger;
+        private readonly IDeltaTransactionRetriever _retriever;
+
+        public TransactionRetrieverStep(IDeltaTransactionRetriever retriever, ILogger logger)
+        {
+            _retriever = retriever ?? throw new ArgumentNullException(nameof(retriever));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name '' is not declared in the argument list.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4979576)